### PR TITLE
[encoding/text_encoding] Add support for text marshaling/unmarshaling with separators

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Add support for text marshaling/unmarshaling with separators

**Link to tracking Issue:**
#32679

**Testing:**
Unit tests.

**Documentation:**
README

### CODE REVIEW

**Review by blakeromano — APPROVED** [review]
lgtm

**Comment by atoulme** [comment]
@open-telemetry/collector-contrib-maintainer Can I please request a review and merge?

File: extension/encoding/textencodingextension/README.md [hunk 1]
```diff
@@ -9,4 +9,20 @@
...
-<!-- end autogenerated section -->
\ No newline at end of file
+<!-- end autogenerated section -->
+
+The `text_encoding` extension is an encoding extension that can unmarshal and marshal logs.
+
+The extension accepts an encoding and separator to unmarshal data as the body of one or more log records.
+The separator accepts regular expressions.
+
+When marshaling logs, the extension will return the body content, separated by a separator.
```
Comment (andrzej-stencel) [inline extension/encoding/textencodingextension/README.md:19]: I think the docs on the marshaling separator and unmarshaling separator could be made a bit clearer, perhaps something that can be amended in a follow-up pull request.

File: extension/encoding/textencodingextension/text.go [hunk 2]
```diff
@@ -13,23 +16,62 @@ import (
...
+	} else {
+		s.Split(func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+			if atEOF && len(data) == 0 {
+				return 0, nil, nil
+			}
+			return len(data), data, nil
+		})
+	}
+	for s.Scan() {
+		l := p.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
```
Comment (andrzej-stencel) [inline extension/encoding/textencodingextension/text.go:51]: Each log is put into its own scope and resource? :thinking:

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**